pygluu/kubernetes/templates/helm/gluu/charts/config/templates/configmaps.yaml

```diff
@@ -25,21 +25,28 @@ data:
   GLUU_CONFIG_ADAPTER: {{ .Values.global.configAdapterName }}
   GLUU_SECRET_ADAPTER: {{ .Values.global.configSecretAdapter }}
   # [google_envs] Envs related to using Google
+  {{ if or (eq .Values.global.configSecretAdapter "google") (eq .Values.global.cnPersistenceType "spanner") }}
   GOOGLE_APPLICATION_CREDENTIALS: {{ .Values.global.cnGoogleApplicationCredentials | quote }}
   GOOGLE_PROJECT_ID: {{ .Values.configmap.cnGoogleProjectId | quote }}
+  {{- end }}
   # [google_spanner_envs] Envs related to using Google Secret Manager to store config and secret layer
+  {{ if eq .Values.global.cnPersistenceType "spanner" }}
   GLUU_GOOGLE_SPANNER_INSTANCE_ID: {{ .Values.configmap.cnGoogleSpannerInstanceId | quote }}
   GLUU_GOOGLE_SPANNER_DATABASE_ID: {{ .Values.configmap.cnGoogleSpannerDatabaseId | quote }}
   SPANNER_EMULATOR_HOST: {{ .Values.configmap.cnGoogleSpannerEmulatorHost | quote }}
   # [google_spanner_envs] END
+  {{- end }}
   # [google_secret_manager_envs] Envs related to using Google Secret Manager to store config and secret layer
+  {{ if eq .Values.global.configSecretAdapter "google" }}
   GLUU_SECRET_GOOGLE_SECRET_VERSION_ID: {{ .Values.configmap.cnSecretGoogleSecretVersionId | quote }}
   GLUU_SECRET_GOOGLE_SECRET_MANAGER_PASSPHRASE: {{ .Values.configmap.cnGoogleSecretManagerPassPhrase | quote }}
   GLUU_SECRET_GOOGLE_SECRET_NAME_PREFIX: {{ .Values.configmap.cnSecretGoogleSecretNamePrefix | quote }}
   GLUU_CONFIG_GOOGLE_SECRET_VERSION_ID: {{ .Values.configmap.cnConfigGoogleSecretVersionId | quote }}
   GLUU_CONFIG_GOOGLE_SECRET_NAME_PREFIX: {{ .Values.configmap.cnConfigGoogleSecretNamePrefix | quote }}
```

Comment:
adjust these to the new name of the envs which are 

`GLUU_GOOGLE_SECRET_VERSION_ID` and `GLUU_GOOGLE_SECRET_NAME_PREFIX` The rest can be dropped and removed from the values.yaml as well.